RUFAS/biophysical/animal/animal_genetics/animal_genetics.py

```diff
@@ -1,317 +1,216 @@
+        mean_tbv_fat = (tbv_fat_top_semen + dam_tbv_fat) / 2
+        mean_tbv_protein = (tbv_protein_top_semen + dam_tbv_protein) / 2
 
-        Returns
-        -------
-        dict[str, dict[str, dict[str, float]]]
-            The net merit data after filling the gap in between entries, $USD.
-        """
-        expanded_net_merit: dict[str, dict[str, dict[str, float]]] = {}
-        monthly_increase_lookup = {2005: 132 / 60, 2010: 184 / 60, 2015: 231 / 60, 2020: (360.731239 - 36.931108) / 48}
+        std_tbv_fat = np.sqrt(std_tbv_fat_national_average**2 / 2)
+        std_tbv_protein = np.sqrt(std_tbv_protein_national_average**2 / 2)
 
-        for breed in original_net_merit.keys():
-            expanded_net_merit[breed] = {}
-            years = [int(year_month[:4]) for year_month in original_net_merit[breed].keys()]
-            max_year = max(years)
-            current_keys = list(original_net_merit[breed].keys())
-            for year_month in current_keys:
-                expanded_net_merit[breed][year_month] = {
-                    "average": original_net_merit[breed][year_month]["average"],
-                    "std": original_net_merit[breed][year_month]["std"],
-                }
-                year, month = int(year_month[:4]), int(year_month[5:])
+        tbv_fat, tbv_protein = Utility.generate_bivariate_random_numbers(
+            mean_tbv_fat, mean_tbv_protein, std_tbv_fat, std_tbv_protein, TBV_CORRELATION
+        )
 
-                if month < 12:
-                    month += 1
-                else:
-                    year += 1
-                    month = 1
-                next_year_month = str(year) + "-" + str(month).zfill(2)
-                num_inc = 1
-                while next_year_month not in current_keys:
-                    average_monthly_increase_key = year - (year % 5)
-                    average_monthly_increase = monthly_increase_lookup[average_monthly_increase_key]
-                    expanded_net_merit[breed][next_year_month] = {
-                        "average": original_net_merit[breed][year_month]["average"]
-                        + num_inc * average_monthly_increase,
-                        "std": original_net_merit[breed][year_month]["std"],
-                    }
-                    if month < 12:
-                        month += 1
-                    else:
-                        year += 1
-                        month = 1
-                    next_year_month = str(year) + "-" + str(month).zfill(2)
-                    num_inc += 1
-                    if year > max_year:
-                        break
+        return tbv_fat, tbv_protein
 
-            updated_keys = list(expanded_net_merit[breed].keys())
-            updated_keys.sort()
-            expanded_net_merit[breed] = {k: expanded_net_merit[breed][k] for k in updated_keys}
-        return expanded_net_merit
+    def _calculate_ep_values(self) -> tuple[float, float]:
+        """Calculate E_permanent values."""
```

Comment:
Maybe for the docstrings try to use the non-abbreviated variable name with the (E_permanent) referenced.
```suggestion
        """Calculate Permanent Environment Effect (E_permanent) values."""
```